### PR TITLE
Howework 1: Basics. NodeJS fundamental theory

### PR DESCRIPTION
**Pull request by NikitaUstsiushenka**
More details for task: [Homework 1](https://epam.sharepoint.com/sites/EPAMNode.jsGlobalMentoringProgram/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FHomework%2FModule%201%2FHomework%201%2Epdf&parent=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FHomework%2FModule%201&p=true&originalPath=aHR0cHM6Ly9lcGFtLnNoYXJlcG9pbnQuY29tLzpiOi9zL0VQQU1Ob2RlLmpzR2xvYmFsTWVudG9yaW5nUHJvZ3JhbS9FZWp6bG02X0dVcEttZHBRRFRzS2NSZ0JFMWx1T0tYV3djQjFuVnBaZjZWRkxnP3J0aW1lPXk3M2NYaUw4MkVn)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "plugins": ["prettier"],
+    "globals": {
+        "Atomics": "readonly",
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+    "rules": {
+        // "prettier/prettier": "error"
```
Comment (ghost) [inline .eslintrc.json]: please remove commented rules

File: task1.2/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  .fromFile(csvFilePath)
+  .subscribe((json) => new Promise((resolve) => resolve(json))
+    .then((csvLine) => {
+      writable.write(`${JSON.stringify(csvLine)}\n`);
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      console.log('Data read and written to file with or without error!');
```
Comment (ghost) [inline task1.2/index.js]: this log is not useful as it doesn't show what really happened (success or not?). I suggest to remove finally block and and logging to `then`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "scripts": {
+    "task1": "nodemon ./task1.1/index.js",
+    "task2": "nodemon ./task1.2/index.js"
+  },
+  "author": "Mikita_Ustsiushenka@epam.com",
+  "devDependencies": {
+    "@babel/cli": "^7.13.14",
+    "@babel/core": "^7.13.15",
+    "@babel/node": "^7.13.13",
+    "@babel/preset-env": "^7.13.15",
```
Comment (ghost) [inline package.json:13]: Currently babel is only installed as a dependence, but not used for code transpilation
Current task is actually a little bit outdated, as Node.js 14+ supports es6 modules by default (if type=module added to package.json), but babel can be also useful for other JS features which are not supported by Node.js for now

Comment (ghost) [reply package.json:13]: so the task is to configure babel as a transpiler for you code

**Review by ghost — CHANGES_REQUESTED** [review]

File: task1.2/index.js (hunk 2 above)
Comment (NikitaUstsiushenka) [reply task1.2/index.js]: @alligator2005 Thanks, updated

File: package.json (hunk 3 above)
Comment (NikitaUstsiushenka) [reply package.json:13]: @alligator2005 Thanks, updated

**Review by ghost — APPROVED** [review]

**Pull request merged**